### ISSUE TITLE
Demo (Newspaper ML example)

### ISSUE THREAD

**Issue by mkhattab940** [hakaru-CS4ZP6/hakaru#34]
In our last meeting Dr. Carette mentioned an example we might be able to base a demo off of.

@JacquesCarette can you post the links here when you get a chance to dig them up? Thank you

**Comment by JacquesCarette** [comment]
When is this due?

**Comment by mkhattab940** [comment]
Our final presentation is on April 24th, and I don't think it needs to be submitted before then.

**Comment by JacquesCarette** [comment]
This is coming - there is indeed a demo associated with this dataset.  I am trying to get the details in a reproducible form.

**Comment by JacquesCarette** [comment]
I've pinged the person who has this information on Friday, and again just now.

**Comment by JacquesCarette** [comment]
Finally got it!  Here:

everything can be done from a single repository:
https://github.com/rjnw/hakaru-benchmarks/

starting from the repo top level directory in repo
```make init
make hakaru
cd runners/
make hkbin
make gmm-hs classes=3 points=10
make gmm-hs-acc classes=3 points=10
cd ../output
racket 3 10```

You'll need racket (another PL) as that's a faster backend than the haskell-based sampler.

**Comment by JacquesCarette** [comment]
I would close this, but I'll wait until someone acknowledges receipt first.

**Comment by JacquesCarette** [comment]
ping!

**Comment by NevTheDev14** [comment]
Received! Thank you Dr. Carette.

**Comment by NevTheDev14** [comment]
Hi Dr. Carette,

I was trying to run the Newspaper demo that you had sent following the instructions you gave us, however when I run make gmm-hs classes=3 points=10 I get the following error:

./hkbin/gmmGibbs ../input/GmmGibbs/3-10 ../testcode/jagssrc ../output/
gmmGibbs: ../input/GmmGibbs/3-10: openFile: does not exist (No such file or directory)
Makefile:9: recipe for target 'gmm-hs' failed
make: *** [gmm-hs] Error 1

So basically GmmGibbs/3-10 is missing.
Is there something I'm suppose to do that generates the 3-10 file?

**Comment by JacquesCarette** [comment]
@rjnw can you give us a hand please?

**Comment by rjnw** [comment]
@JacquesCarette  Ken told me that what you want is NaiveBayes nor the gmmGibbs right.
The example above is for gmmGibbs and requires the synthetic data to be generated which can be generated in `input` directory by running `make`. 

For running newsgroup:
in file `hakaru-benchmarks/runners/hk/NaiveBayesGibbs/Main.hs`
on line 30 change numTrials to 1
comment out line 33,34,36,43,44,45.
on 47 change minSeconds to 1
and on line 49 change minSweeps to 1
go to directory runners and do the following
```
make hkbin
./hkbin/naiveBayesGibbs ../input ../testcode/jagssrc ../output
```
the output is in `output/NaiveBayesGibbs/NaiveBayesGibbs/hk/20-19997 file`

**Comment by JacquesCarette** [comment]
@rjnw Actually what they wanted (their presentation is in 30 minutes...) was an example they could use to show all of this actually running.  The details don't so much matter, as much as being an example that is easy to explain and understand.

Such an example would be good to have for Hakaru in general.

**Comment by JacquesCarette** [comment]
re-closing, as long obsolete.
